OriginProtocol/origin
# send form only dataURis

### Description:
[Issue # 1112](https://github.com/OriginProtocol/origin/issues/1112)

**Problem:**
- While creating a listing, if a user uploaded two image or more and changed the order of the images, they would receive an error. This was because the form was submitted with an array of objects (with keys of `originalImageFile` and `croppedImageUri`. Form submission expects an array of dataURi strings.

**Solution:**
- Use the photo-picker method `picURIsOnly` to only get the dataURi for the images when `onFormDataChange` is called in `origin-dapp/src/components/listing-create.js`.

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] Wrap any displayed ETH addresses with [`formattedAddress`](https://github.com/OriginProtocol/origin/blob/master/origin-dapp/src/utils/user.js#L15-L17)
- [ ] Wrap any new text/strings for translation
- [ ] Map any new environment variables with a default value in the Webpack config
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)

### Manual Testing:
**Note** There is a current issue w/ react-overlays, and so to manually test you will most likely have to do the following in the command line `origin` root:
```
lerna clean
rm -rf node_modules
npm install
cd node_modules/react-bootstrap
npm install react-overlays@1.0.0-beta.17
cd ../..
npm start
```
1. Go to the create a listing flow, and fill out the required details (Title etc)
2. Upload at least two images.
3. Change the order of the images (they are draggable)
4. When you press `Continue`, you should not encounter errors.

**Note** When the issue was first reported, the user uploaded 10 images and sorted them.
